web/src/components/storage/SearchedDeviceMenu.tsx

```diff
@@ -25,27 +25,24 @@ import { Split, Flex, Label } from "@patternfly/react-core";
 import MenuButton, { MenuButtonItem } from "~/components/core/MenuButton";
 import MenuDeviceDescription from "./MenuDeviceDescription";
 import NewVgMenuOption from "./NewVgMenuOption";
-import { useAvailableDrives, useLongestDiskTitle } from "~/hooks/storage/system";
-import { useDrive, useModel } from "~/queries/storage/config-model";
-import { useDrive as useDriveModel } from "~/hooks/storage/drive";
+import { useCandidateDevices, useLongestDiskTitle } from "~/hooks/storage/system";
+import { useModel } from "~/hooks/storage/model";
+import { useDeleteSearched, useSwitchSearched } from "~/hooks/storage/search";
 import * as driveUtils from "~/components/storage/utils/drive";
 import { deviceBaseName, deviceLabel, formattedPath } from "~/components/storage/utils";
 import * as model from "~/types/storage/model";
 import { StorageDevice } from "~/types/storage";
 import { sprintf } from "sprintf-js";
 import { _, formatList } from "~/i18n";
 
-const driveBaseName = (device: StorageDevice): string => deviceBaseName(device, true);
-const driveLabel = (device: StorageDevice): string => deviceLabel(device, true);
+const baseName = (device: StorageDevice): string => deviceBaseName(device, true);
+const label = (device: StorageDevice): string => deviceLabel(device, true);
 
-const UseOnlyOneOption = (drive: model.Drive): boolean => {
-  const driveModel = useDrive(drive.name);
-  if (!driveModel) return false;
+const UseOnlyOneOption = (device: model.Drive | model.MdRaid): boolean => {
+  const hasPv = device.isTargetDevice;
+  if (!driveUtils.hasFilesystem(device) && (hasPv || device.isExplicitBoot)) return true;
```

Comment:
`#hasFilesystem` is something we can move from the utils to the model. BTW, `hasFilesystem` accepts a `apiModel.Drive` but `device` here has other type.